From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Wed, 5 Aug 2015 02:00:56 -0400
Subject: 9035-qemu-snapshot-save-persistent-domain-config

commit 9036b31aeddb63db198576b8eaba331df105c0c6
Author: Peter Krempa <pkrempa@redhat.com>
Date:   Mon Jun 30 13:44:26 2014 +0200

    qemu: snapshot: Save persistent domain config when taking external snapshot
    
    Commit 55bbb011b965c7962933604c70f61cef45e8ec04 introduced a regression
    where we forgot to save the persistent domain configuration after an
    external snapshot. This would make libvirt forget the snapshots and
    effectively revert to the previous state in the following scenario:
    
    1) Start VM
    2) Take snapshot
    3) Destroy VM
    4) Restart libvirtd
    
    Also fix spurious blank line added by patch mentioned above.
---
 src/qemu/qemu_driver.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/qemu/qemu_driver.c b/src/qemu/qemu_driver.c
index 10feaf6..761ea9b 100644
--- a/src/qemu/qemu_driver.c
+++ b/src/qemu/qemu_driver.c
@@ -12970,8 +12970,10 @@ qemuDomainSnapshotCreateDiskActive(virQEMUDriverPtr driver,
             int indx = virDomainDiskIndexByName(vm->newDef,
                                                 vm->def->disks[i]->dst,
                                                 false);
-            if (indx >= 0)
+            if (indx >= 0) {
                 persistDisk = vm->newDef->disks[indx];
+                persist = true;
+            }
         }
 
         ret = qemuDomainSnapshotCreateSingleDiskActive(driver, vm,
@@ -13017,7 +13019,6 @@ qemuDomainSnapshotCreateDiskActive(virQEMUDriverPtr driver,
                         persistDisk = vm->newDef->disks[indx];
                         persist = true;
                     }
-
                 }
 
                 qemuDomainSnapshotUndoSingleDiskActive(driver, vm,
